src/schema.cpp

```diff
@@ -148,17 +148,32 @@ const char* MODELS_CONFIG_SCHEMA = R"({
 		},
 		"node_config": {
 			"type": "object",
-			"required": ["name", "model_name", "inputs", "outputs"],
+			"required": ["name", "type", "inputs", "outputs"],
+			"oneOf": [
+    			{
+        			"properties": { "type": { "enum": ["custom"] } },
+        			"required": ["library_name"],
+					"not": { "required": ["model_name"] }
+    			},
+    			{
+        			"properties": { "type": { "enum": ["DL model"] } },
+        			"not": { "required": ["library_name"] },
+					"required": ["model_name"]
+    			}
+  			],
 			"properties": {
 				"name": {
 					"type": "string"
 				},
 				"model_name": {
 					"type": "string"
 				},
+				"library_name": {
+					"type": "string"
+				},
 				"type": {
 					"type": "string",
-					"enum": ["DL model", "Demultiplexer", "Batch dispatcher"]
+					"enum": ["DL model", "Demultiplexer", "Batch dispatcher", "custom"]
```

Comment:
We can remove `Demultiplexer` and `Batch dispatcher`. This is never used (most likely somebody added it thinking ahead). Demultiplexer will be solved without adding new types of nodes.